internal/cmd/testatlas/atlas_test.go

```diff
@@ -8,56 +8,161 @@ package main
-			ApplyURI(uri).
-			SetServerSelectionTimeout(1 * time.Second)
-
-		// Run basic connectivity test.
-		if err := runTest(ctx, clientOpts); err != nil {
-			t.Fatalf("error running test with TLS at index %d: %v", idx, err)
-		}
-
-		tlsConfigSkipVerify := clientOpts.TLSConfig
-		tlsConfigSkipVerify.InsecureSkipVerify = true
-
-		// Run the connectivity test with InsecureSkipVerify to ensure SNI is done correctly even if verification is
-		// disabled.
-		clientOpts.SetTLSConfig(tlsConfigSkipVerify)
-
-		if err := runTest(ctx, clientOpts); err != nil {
-			t.Fatalf("error running test with tlsInsecure at index %d: %v", idx, err)
-		}
+	cases := []struct {
+		name        string
+		envVar      string
+		certKeyFile string
+		wantErr     string
+	}{
+		{
+			name:        "Atlas with TLS",
+			envVar:      "ATLAS_REPL",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with TLS and shared cluster",
+			envVar:      "ATLAS_SHRD",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with free tier",
+			envVar:      "ATLAS_FREE",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with TLS 1.1",
+			envVar:      "ATLAS_TLS11",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with TLS 1.2",
+			envVar:      "ATLAS_TLS12",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with serverless",
+			envVar:      "ATLAS_SERVERLESS",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on replica set",
+			envVar:      "ATLAS_SRV_REPL",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on shared cluster",
+			envVar:      "ATLAS_SRV_SHRD",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on free tier",
+			envVar:      "ATLAS_SRV_FREE",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on TLS 1.1",
+			envVar:      "ATLAS_SRV_TLS11",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on TLS 1.2",
+			envVar:      "ATLAS_SRV_TLS12",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with srv file on serverless",
+			envVar:      "ATLAS_SRV_SERVERLESS",
+			certKeyFile: "",
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with X509 Dev",
+			envVar:      "ATLAS_X509_DEV",
+			certKeyFile: createAtlasX509DevCertKeyFile(t),
+			wantErr:     "",
+		},
+		{
+			name:        "Atlas with X509 Dev no user",
+			envVar:      "ATLAS_X509_DEV",
+			certKeyFile: createAtlasX509DevCertKeyFileNoUser(t),
+			wantErr:     "UserNotFound",
+		},
 	}
 
-	t.Logf("Finished!")
+	for _, tc := range cases {
+		t.Run(fmt.Sprintf("%s (%s)", tc.name, tc.envVar), func(t *testing.T) {
+			uri := os.Getenv(tc.envVar)
+			if uri == "" {
+				t.Skipf("Environment variable %q is not set", tc.envVar)
+			}
```

Comment:
Skipping if the env vars aren't set will cause the tests to silently pass if the env vars somehow get unset, even if there is a bug. Is there a way to not skip in this case?

Reply:
Sounds like a great solution!